### ISSUE TITLE
commit  #7344 (Fix in CarrierFactory) not present in v2.1.5

### ISSUE THREAD

**Issue by surajsanchit** [magento/magento2#9175]
Commit  #7344 Is not present in release 2.1.5 which causes shipping modules depending on storeview specific configurations not to behave as expected.

Seeing as the fix is merged on 07-11-2016 I would expect it to be available in the latest stable release.

**Comment by korostii** [comment]
Yep the commit 08b5177d6745596f2993400e8a6a9a6ada69308f is only present on branch develop http://take.ms/NyoJ1 meaning it will normally appear in 2.2 unless additional actions are taken. 

I am not sure if the Magento, Inc. has this additional step (backport any small bugfix into 2.0 and 2.1 branches as soon as it's fixed on develop) in their internal workflow. That would certainly be something a lot of people would benefit from, but unfortunately it is unlikely that the bugs "fixed in develop but not on 2.1" are even tracked anywhere.

**Comment by surajsanchit** [comment]
Wow thats a mindfuck, so hotfixes/bugfixes arent released in patches at all (2.1.xx)? Anyway we can speed this up, e.a. any forks/releases that include the develop bugfixes? 

Without the fixes its really cumbersome running Magento 2 on production enviroments

**Comment by korostii** [comment]
Well most of the fixes are on the develop branch, but running a production instance with code from this branch is even more risky since it also contains unfinished features. You could make a fork, and cherry-pick those fixes one-by-one, but then again such a flow would require a lot of effort and defeat the usage of composer as such. For a small development team the "fix the most critical ones via di.xml" approach might just be the most cost-effective.
Honestly, I don't see any straightforward workaround. I would suggest that they (Magento, Inc.) just need to release small patchy 2.1.x.x versions each week, but it seems that their current release process just cannot accommodate the speed, as each release goes through a long queue of tests and what-else.
Basically, that's also why we keep seeing some bugs reported here up 10 times each: they just drag on for months unfixed, waiting for their time to come.

There is one way we can speed it up though: if it's important for you enough, you could make a fork and produce a pull request targeted to the 2.1 branch (or rather 2.1-develop, not sure about this part). Making a pull request currently seems to get faster attention from core developers.

**Comment by veloraven** [comment]
@surajsanchit please format this issue according to the [Issue reporting guidelines](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines): with steps to reproduce, actual result and expected result.

**Comment by korostii** [comment]
@veloraven, this is a request to backport a fix which is already accepted into the `develop` branch.
Please read the description provided with the pull request referenced in the report if you require any more details: https://github.com/magento/magento2/pull/7344

**Comment by PascalBrouwers** [comment]
I'm starting to suspect more and more than veloraven is just a bot.

**Comment by Ctucker9233** [comment]
@PascalBrouwers I believe you are right.

**Comment by magento-engcom-team** [comment]
@surajsanchit thank you for your report.
Please use the issue [reporting guidelines](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines) to report an issue. Please, edit issue description and include the preconditions, the steps required to reproduce, the actual result, and the expected result.
Please, add the following information:
 - [ ] Preconditions: Magento version
 - [ ] Steps to reproduce
 - [ ] Expected result
 - [ ] Actual result

The issue does not exist in 2.2.0 which will be available soon.

@PascalBrouwers thank you for being a part of Magento Community where everyone follow [Code of Conduct](https://github.com/magento/magento2/blob/develop/.github/CODE_OF_CONDUCT.md)

**Comment by PascalBrouwers** [comment]
Que?
..looks up Code of Conduct to see if he missed something.

**Comment by korostii** [comment]
It's probably assumed to be something vague in between "Trolling, insulting/derogatory comments, and personal or political attacks" and not "Showing empathy towards other community members".

Well guess what, forcing someone to edit the issue description weeks (which is already provided in the linked PR #7344 ) weeks after the fact - that isn't too empathetic neither.

Speaking of "Code of Conduct", my previous message was clearly ignored. 
That's not respectful of my differing viewpoint.  
That's not "Gracefully accepting constructive criticism". 
That's **not** "Focusing on what is best for the community".

**Comment by magento-engcom-team** [comment]
@surajsanchit we are closing this issue due to inactivity. If you'd like to update it, please reopen the issue.
